core-primitives/sgx/temp-dir/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+//! # temp-dir
+//!
+//! Copied from the original tempdir crate with tiny adjustments for SGX-compatibility.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use core::sync::atomic::{AtomicU32, Ordering};
+use std::{
+	borrow::ToOwned,
+	collections::hash_map::RandomState,
+	format,
+	hash::{BuildHasher, Hasher},
+	path::{Path, PathBuf},
+	string::String,
+};
+
+/// Serve some low-security random ID to prevent temp-dir clashes across multiple processes.
+fn rand_id() -> String {
+	// u64 always has more than 4 bytes so this never panics.
+	format!("{:x}", RandomState::new().build_hasher().finish())[..4].to_owned()
+}
+
+lazy_static::lazy_static! {
+	/// A unique identifier, which is instanciated upon process start, but it is
+	/// not the process id itself.
+	///
+	/// This is a workaround for `sgx_tstd` lib not exposing the `process::id()`.
+	pub static ref PROCESS_UNIQUE_ID: String = rand_id();
+}
+
+static COUNTER: AtomicU32 = AtomicU32::new(0);
+
+/// The path of an existing writable directory in a system temporary directory.
+///
+/// Drop the struct to delete the directory and everything under it.
+/// Deletes symbolic links and does not follow them.
+///
+/// Ignores any error while deleting.
+/// See [`TempDir::panic_on_cleanup_error`](struct.TempDir.html#method.panic_on_cleanup_error).
+///
+/// # Example
+/// ```rust
+/// use itp_sgx_temp_dir::TempDir;
+/// let d = TempDir::new().unwrap();
+/// // Prints "/tmp/t1a9b-0".
+/// println!("{:?}", d.path());
+/// let f = d.child("file1");
+/// // Prints "/tmp/t1a9b-0/file1".
+/// println!("{:?}", f);
+/// std::fs::write(&f, b"abc").unwrap();
+/// assert_eq!(
+///     "abc",
+///     std::fs::read_to_string(&f).unwrap(),
+/// );
+/// // Prints "/tmp/t1a9b-1".
+/// println!("{:?}", TempDir::new().unwrap().path());
+/// ```
```

Comment:
I think example comments are useful but this one could use just a tiny refactor for readability.

Reply:
Haha, this one is just copy-pasted from the original crate. With this argument, I shall leave it at that.